### PR TITLE
Add GMX GLP on Avalanche

### PR DESCRIPTION
**Pull request by theachenyj**
Brief comments on the purpose of your changes:
Created models /gmx/avalanche_c/glp , and I added 2 spells in this folder.

**For Dune Engine V2**

I've checked that:

### General checks:
* [ ] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [ ] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [ ] if adding a new model, I added a test
* [ ] the filename is unique and ends with .sql
* [ ] each sql file is a select statement and has only one view, table or function defined  
* [ ] column names are `lowercase_snake_cased`
* [ ] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [ ] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributor Dune usernames)

### Pricing checks:
* [ ] `coin_id` represents the ID of the coin on coinpaprika.com
* [ ] all the coins are active on coinpaprika.com (please remove inactive ones)

### Join logic:
* [ ] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

### Incremental logic:
* [ ] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [ ] where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

File: models/gmx/avalanche_c/glp/gmx_avalanche_c_glp_schema.yml [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+        description: "Net asset under management for the WETH.e pool"
+        tests:
+          - not_null
+      - &wbtc_e_aum
+        name: wbtc_e_aum
+        description: "Net asset under management for the WBTC.e pool"
+        tests:
+          - not_null
+      - &btc_b_aum
+        name: wbtc_aum
```
Comment (Hosuke) [inline models/gmx/avalanche_c/glp/gmx_avalanche_c_glp_schema.yml:193]: Hi @theachenyj, you may have a typo here, do you mean `btc_b_aum`.

Comment (theachenyj) [reply models/gmx/avalanche_c/glp/gmx_avalanche_c_glp_schema.yml]: Nice catch! I'd committed a fix. Thx.

**Comment by Hosuke** [comment]
Hi @theachenyj , since these are some new models, could you possibly make a seed file with a few records like [this](https://github.com/duneanalytics/spellbook/blob/main/seeds/jarvis_network/polygon/jarvis_network_polygon_view_transactions_seed.csv) for the `glp_components` and the `glp_aum` model?

**Comment by theachenyj** [comment]
> Hi @theachenyj , since these are some new models, could you possibly make a seed file with a few records like [this](https://github.com/duneanalytics/spellbook/blob/main/seeds/jarvis_network/polygon/jarvis_network_polygon_view_transactions_seed.csv) for the `glp_components` and the `glp_aum` model?

Since the data is a aggregated result, it's not easy to find exact comparison. Besides the result would  be slightly different with different calculation method. I think I can get some seed datas from [official web site](https://stats.gmx.io/avalanche) , but in a different granularity, and compare them like this: if the difference between them within a certain value, like 0.1%, I'd say the result is correct. Do you think this way works out?

**Comment by Hosuke** [comment]
> > Hi @theachenyj , since these are some new models, could you possibly make a seed file with a few records like [this](https://github.com/duneanalytics/spellbook/blob/main/seeds/jarvis_network/polygon/jarvis_network_polygon_view_transactions_seed.csv) for the `glp_components` and the `glp_aum` model?
> 
> Since the data is a aggregated result, it's not easy to find exact comparison. Besides the result would be slightly different with different calculation method. I think I can get some seed datas from [official web site](https://stats.gmx.io/avalanche) , but in a different granularity, and compare them like this: if the difference between them within a certain value, like 0.1%, I'd say the result is correct. Do you think this way works out?

Yes, for comparison between double typed values we can accept a range, and usually we use discrete/string typed value like addresses in the seed for test. So do not worry the seed record need be exactly matched to the query result.

**Comment by Hosuke** [comment]
Currently the [test for glp_aum fails](https://github.com/duneanalytics/spellbook/actions/runs/3599401637/jobs/6063005139#step:12:93), we can remove the test for doubled value first, or accept a larger range.

**Comment by theachenyj** [comment]
> Currently the [test for glp_aum fails](https://github.com/duneanalytics/spellbook/actions/runs/3599401637/jobs/6063005139#step:12:93), we can remove the test for doubled value first, or accept a larger range.

I fix some 

> Currently the [test for glp_aum fails](https://github.com/duneanalytics/spellbook/actions/runs/3599401637/jobs/6063005139#step:12:93), we can remove the test for doubled value first, or accept a larger range.

I'm stuck in the test. It's weird. I've run the test SQL on Dune and it has passed. I have no idea which part has problem.

File: seeds/gmx/avalanche_c/gmx_glp_aum_seed.csv [hunk 2]
```diff
@@ -0,0 +1,6 @@
+date,usdc_aum,usdc_e_aum
+2022-09-15,20359495.092885,18949462.097249
+2022-09-16,20327050.875974,18440021.732571
```
Comment (Hosuke) [inline seeds/gmx/avalanche_c/gmx_glp_aum_seed.csv:3]: Hi @theachenyj, please have a look on this line.
Moreover this is your [test query result](https://dune.com/queries/1707843?d=1)
image
It seems there is a null value for this record when joining.

Comment (theachenyj) [reply seeds/gmx/avalanche_c/gmx_glp_aum_seed.csv:3]: Finally! I've made a mistake in aum cte. Thx a lot.

**Review by Hosuke — APPROVED** [review]
To whom it may concern, I have reviewed this PR and here is the summary:

- Metadata & naming conventions look good ✅
- Config blocks match uniqueness tests in schema files ✅
- Tests and seeds added for the new models ✅
- Incremental filters added when joining large tables ✅
- Contract addresses are lowercased ✅ 

This PR contains 2 new models:
- `gmx_avalanche_c_glp_components`
- `gmx_avalanche_c_glp_aum`

Both models have been modified to incremental views.
Since the seed checking columns are double-typed, the generic test is not applied to these models.

**Comment by jeff-dude** [comment]
thank you both. please do note final commits:
- remove redundant unique test
- add seed to project file, for schema naming standards mostly

**Review by jeff-dude — APPROVED** [review]

**Pull request merged**